diff -uNr Makefile.kernel Makefile.kernel
--- Makefile.kernel	2006-09-03 19:55:36.000000000 +0100
+++ Makefile.kernel	2006-09-03 19:55:52.000000000 +0100
@@ -8,7 +8,7 @@
 
 vm_check_build = $(shell if $(CC) $(CPPFLAGS) $(CFLAGS) $(CFLAGS_KERNEL) \
 	$(EXTRA_CFLAGS) -Iinclude2/asm/mach-default -DKBUILD_BASENAME=\"$(DRIVER)\" \
-	-Werror -S -o /dev/null -xc $(1) \
+	-Wno-declaration-after-statement -Werror -S -o /dev/null -xc $(1) \
 	> /dev/null 2>&1; then echo "$(2)"; else echo "$(3)"; fi)
 
 CC_WARNINGS := -Wall -Wstrict-prototypes 
