--- Makefile.kernel
+++ Makefile.kernel
@@ -14,7 +14,7 @@
 endif
 vm_check_build = $(shell if $(CC) $(CPPFLAGS) $(CFLAGS) $(CFLAGS_KERNEL) \
 	$(EXTRA_CFLAGS) -Iinclude2/asm/mach-default -DKBUILD_BASENAME= \
-	-Werror -S -o /dev/null -xc $(1) \
+	-Werror -Wno-unused -S -o /dev/null -xc $(1) \
 	$(VMWARE_DEBUG_REDIRECT); then echo "$(2)"; else echo "$(3)"; fi)
 
 CC_WARNINGS := -Wall -Wstrict-prototypes 
