diff --git a/vsock-only/Makefile.kernel b/vsock-only/Makefile.kernel
index 69326f6..0904ef2 100644
--- a/vsock-only/Makefile.kernel
+++ b/vsock-only/Makefile.kernel
@@ -7,12 +7,6 @@
 ####
 ####
 
-vm_check_build = $(shell if $(CC) $(CPPFLAGS) $(CFLAGS) $(CFLAGS_KERNEL) \
-	$(EXTRA_CFLAGS) -Iinclude2/asm/mach-default \
-	-DKBUILD_BASENAME=\"$(DRIVER)\" \
-	-Werror -S -o /dev/null -xc $(1) \
-	> /dev/null 2>&1; then echo "$(2)"; else echo "$(3)"; fi)
-
 vm_product_defines = $(if $(findstring tools,$(1)), -DVMX86_TOOLS,)
 
 CC_WARNINGS := -Wall -Wstrict-prototypes
