diff -uNr vmmon-only/include/iocontrols_compat.h vmmon-only/include/iocontrols_compat.h
--- vmmon-only/include/iocontrols_compat.h	2007-07-23 10:59:47.000000000 +0100
+++ vmmon-only/include/iocontrols_compat.h	2007-12-01 15:55:10.000000000 +0000
@@ -15,7 +15,7 @@
 #define VMMON_VERSION_V5        (116 << 16 | 0)
 #define VMMON_VERSION_V55       (137 << 16 | 0)
 #define VMMON_VERSION_S1B1      (138 << 16 | 0)
-#define VMMON_VERSION_V6        (161 << 16 | 0)
+#define VMMON_VERSION_V6        (167 << 16 | 0)
 
 /*
  * Limit structure for CREATE_VM
diff -uNr vmmon-only.old/include/vcpuset.h vmmon-only/include/vcpuset.h
--- vmmon-only.old/include/vcpuset.h	2007-07-22 12:21:17.000000000 +0100
+++ vmmon-only/include/vcpuset.h	2007-12-01 15:52:11.000000000 +0000
@@ -71,7 +71,7 @@
 #define ffs(x) ffs_x86_64(x)
 
 #elif defined MODULE
-   #include "asm/bitops.h"
+   #include "linux/bitops.h"
 #elif defined __APPLE__ && defined KERNEL
    /* 
     * XXXMACOS An ugly hack to resolve redefinition of PAGE_ defines 
