--- vmblock-only/linux/vmblockInt.h       2008-08-07 20:56:05.000000000 +0200
+++ vmblock-only/linux/vmblockInt.h       2008-08-07 21:00:10.000000000 +0200
@@ -50,7 +50,16 @@
 #endif /* __KERNEL__ */

 #define VMBLOCK_CONTROL_MODE       S_IRUSR | S_IFREG
-#define VMBLOCK_CONTROL_PARENT     proc_root_fs
+/*
+ * constants like proc_root_fs has been removed from proc_fs.h
+ * since kernel 2.6.26. We have to replace it by NULL and use
+ * full path : to create /proc/fs/wonderful => use fs/wonderful
+ */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 26)
+#  define VMBLOCK_CONTROL_PARENT     proc_root_fs
+#else
+#  define VMBLOCK_CONTROL_PARENT     NULL
+#endif

 /*
  * Our modules may be compatible with kernels built for different processors.
--- vmblock-only/include/vmblock.h        2008-05-16 07:59:29.000000000 +0200
+++ vmblock-only/include/vmblock.h        2008-08-07 21:09:13.000000000 +0200
@@ -14,9 +14,16 @@
 #ifdef sun
 # include <sys/ioccom.h>
 #endif
-
-#define VMBLOCK_FS_NAME                "vmblock"
-
+/*
+ * constants like proc_root_fs has been removed from proc_fs.h
+ * since kernel 2.6.26. We have to replace it by NULL and use
+ * full path : to create /proc/fs/wonderful => use fs/wonderful
+ */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 26)
+# define VMBLOCK_FS_NAME                "vmblock"
+#else
+# define VMBLOCK_FS_NAME                "fs/vmblock"
+#endif
 /* Commands for the control half of vmblock driver */
 #if defined(linux)
 # define VMBLOCK_ADD_FILEBLOCK          98

