diff --git a/vmci-only/include/compat_semaphore.h b/vmci-only/include/compat_semaphore.h
index f5527b9..bc37afd 100644
--- a/vmci-only/include/compat_semaphore.h
+++ b/vmci-only/include/compat_semaphore.h
@@ -45,5 +45,13 @@
    #endif
 #endif
 
+#if (defined CONFIG_PREEMPT_RT && LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 31)) || LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 37)
+   #ifndef DECLARE_MUTEX
+      #define DECLARE_MUTEX(_m)  DEFINE_SEMAPHORE(_m)
+   #endif
+   #ifndef init_MUTEX
+      #define init_MUTEX(_m) sema_init(_m,1)
+   #endif
+#endif
 
 #endif /* __COMPAT_SEMAPHORE_H__ */
diff --git a/vmmon-only/include/compat_semaphore.h b/vmmon-only/include/compat_semaphore.h
index f5527b9..bc37afd 100644
--- a/vmmon-only/include/compat_semaphore.h
+++ b/vmmon-only/include/compat_semaphore.h
@@ -45,5 +45,13 @@
    #endif
 #endif
 
+#if (defined CONFIG_PREEMPT_RT && LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 31)) || LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 37)
+   #ifndef DECLARE_MUTEX
+      #define DECLARE_MUTEX(_m)  DEFINE_SEMAPHORE(_m)
+   #endif
+   #ifndef init_MUTEX
+      #define init_MUTEX(_m) sema_init(_m,1)
+   #endif
+#endif
 
 #endif /* __COMPAT_SEMAPHORE_H__ */
diff --git a/vmnet-only/compat_semaphore.h b/vmnet-only/compat_semaphore.h
index f5527b9..548782d 100644
--- a/vmnet-only/compat_semaphore.h
+++ b/vmnet-only/compat_semaphore.h
@@ -45,5 +45,14 @@
    #endif
 #endif
 
+#if (defined CONFIG_PREEMPT_RT && LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 31)) || LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 37)
+   #ifndef DECLARE_MUTEX
+      #define DECLARE_MUTEX(_m)  DEFINE_SEMAPHORE(_m)
+   #endif
+   #ifndef init_MUTEX
+      #define init_MUTEX(_m) sema_init(_m,1)
+   #endif
+#endif
+
 
 #endif /* __COMPAT_SEMAPHORE_H__ */
diff --git a/vmnet-only/vnetFilter.h b/vmnet-only/vnetFilter.h
index fe5c71f..303009f 100644
--- a/vmnet-only/vnetFilter.h
+++ b/vmnet-only/vnetFilter.h
@@ -203,5 +203,13 @@ struct VNet_SetLogLevel {
 } 
 #include "vmware_pack_end.h"
 VNet_SetLogLevel;
+#if (defined CONFIG_PREEMPT_RT && LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 31)) || LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 37)
+   #ifndef DECLARE_MUTEX
+      #define DECLARE_MUTEX(_m)  DEFINE_SEMAPHORE(_m)
+   #endif
+   #ifndef init_MUTEX
+      #define init_MUTEX(_m) sema_init(_m,1)
+   #endif
+#endif
 
 #endif // ifndef _VNETFILTER_H_
diff --git a/vsock-only/linux/af_vsock.h b/vsock-only/linux/af_vsock.h
index 3a8a0ce..d36b2ae 100644
--- a/vsock-only/linux/af_vsock.h
+++ b/vsock-only/linux/af_vsock.h
@@ -88,4 +88,14 @@ typedef struct VSockVmciSock {
 #endif
 } VSockVmciSock;
 
+#if (defined CONFIG_PREEMPT_RT && LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 31)) || LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 37)
+   #ifndef DECLARE_MUTEX
+      #define DECLARE_MUTEX(_m)  DEFINE_SEMAPHORE(_m)
+   #endif
+   #ifndef init_MUTEX
+      #define init_MUTEX(_m) sema_init(_m,1)
+   #endif
+#endif
+
+
 #endif /* __AF_VSOCK_H__ */
