diff -rupN vmblock-only/shared/vm_assert.h vmblock-only.new/shared/vm_assert.h
--- vmblock-only/shared/vm_assert.h	2013-11-05 23:33:27.000000000 -0500
+++ vmblock-only.new/shared/vm_assert.h	2014-10-07 18:13:59.583815971 -0400
@@ -237,12 +237,12 @@ EXTERN void WarningThrottled(uint32 *cou
 #define LOG_ONCE(_s) DO_ONCE(Log _s)
 
 #ifdef VMX86_DEVEL
-   #define DEPRECATED(_fix) DO_ONCE(                                        \
+   #define VMWARE_DEPRECATED(_fix) DO_ONCE(                                        \
                                Warning("%s:%d: %s is DEPRECATED; %s\n",     \
                                           __FILE__, __LINE__, __FUNCTION__, \
                                           _fix))
 #else
-   #define DEPRECATED(_fix) do {} while (0)
+   #define VMWARE_DEPRECATED(_fix) do {} while (0)
 #endif
 
 
diff -rupN vmci-only/shared/vm_assert.h vmci-only.new/shared/vm_assert.h
--- vmci-only/shared/vm_assert.h	2013-11-05 23:33:27.000000000 -0500
+++ vmci-only.new/shared/vm_assert.h	2014-10-07 18:14:18.470816995 -0400
@@ -237,12 +237,12 @@ EXTERN void WarningThrottled(uint32 *cou
 #define LOG_ONCE(_s) DO_ONCE(Log _s)
 
 #ifdef VMX86_DEVEL
-   #define DEPRECATED(_fix) DO_ONCE(                                        \
+   #define VMWARE_DEPRECATED(_fix) DO_ONCE(                                        \
                                Warning("%s:%d: %s is DEPRECATED; %s\n",     \
                                           __FILE__, __LINE__, __FUNCTION__, \
                                           _fix))
 #else
-   #define DEPRECATED(_fix) do {} while (0)
+   #define VMWARE_DEPRECATED(_fix) do {} while (0)
 #endif
 
 
diff -rupN vmmon-only/include/vm_assert.h vmmon-only.new/include/vm_assert.h
--- vmmon-only/include/vm_assert.h	2013-11-06 00:40:51.000000000 -0500
+++ vmmon-only.new/include/vm_assert.h	2014-10-07 18:14:12.118816650 -0400
@@ -237,12 +237,12 @@ EXTERN void WarningThrottled(uint32 *cou
 #define LOG_ONCE(_s) DO_ONCE(Log _s)
 
 #ifdef VMX86_DEVEL
-   #define DEPRECATED(_fix) DO_ONCE(                                        \
+   #define VMWARE_DEPRECATED(_fix) DO_ONCE(                                        \
                                Warning("%s:%d: %s is DEPRECATED; %s\n",     \
                                           __FILE__, __LINE__, __FUNCTION__, \
                                           _fix))
 #else
-   #define DEPRECATED(_fix) do {} while (0)
+   #define VMWARE_DEPRECATED(_fix) do {} while (0)
 #endif
 
 
diff -rupN vmnet-only/vm_assert.h vmnet-only.new/vm_assert.h
--- vmnet-only/vm_assert.h	2013-11-06 00:40:52.000000000 -0500
+++ vmnet-only.new/vm_assert.h	2014-10-07 18:14:05.167816273 -0400
@@ -237,12 +237,12 @@ EXTERN void WarningThrottled(uint32 *cou
 #define LOG_ONCE(_s) DO_ONCE(Log _s)
 
 #ifdef VMX86_DEVEL
-   #define DEPRECATED(_fix) DO_ONCE(                                        \
+   #define VMWARE_DEPRECATED(_fix) DO_ONCE(                                        \
                                Warning("%s:%d: %s is DEPRECATED; %s\n",     \
                                           __FILE__, __LINE__, __FUNCTION__, \
                                           _fix))
 #else
-   #define DEPRECATED(_fix) do {} while (0)
+   #define VMWARE_DEPRECATED(_fix) do {} while (0)
 #endif
 
 
diff -rupN vsock-only/shared/vm_assert.h vsock-only.new/shared/vm_assert.h
--- vsock-only/shared/vm_assert.h	2013-11-05 23:33:27.000000000 -0500
+++ vsock-only.new/shared/vm_assert.h	2014-10-07 18:14:24.454817319 -0400
@@ -237,12 +237,12 @@ EXTERN void WarningThrottled(uint32 *cou
 #define LOG_ONCE(_s) DO_ONCE(Log _s)
 
 #ifdef VMX86_DEVEL
-   #define DEPRECATED(_fix) DO_ONCE(                                        \
+   #define VMWARE_DEPRECATED(_fix) DO_ONCE(                                        \
                                Warning("%s:%d: %s is DEPRECATED; %s\n",     \
                                           __FILE__, __LINE__, __FUNCTION__, \
                                           _fix))
 #else
-   #define DEPRECATED(_fix) do {} while (0)
+   #define VMWARE_DEPRECATED(_fix) do {} while (0)
 #endif
 
 
