--- vmmon-only/linux/hostif.c	2017-05-06 16:56:09.000000000 +0200
+++ vmmon-only/linux/hostif.c.new	2017-05-06 16:56:41.000000000 +0200
@@ -73,6 +73,7 @@
 #include <linux/capability.h>
 #include <linux/kthread.h>
 #include <linux/wait.h>
+#include <linux/sched/signal.h>
 
 #include <asm/apic.h>
 
--- vmnet-only/userif.c	2017-03-13 04:05:59.000000000 +0100
+++ vmnet-only/userif.c.new	2017-05-06 16:58:32.000000000 +0200
@@ -36,6 +36,7 @@
 #include <linux/slab.h>
 #include <linux/version.h>
 #include <linux/wait.h>
+#include <linux/sched/signal.h>
 
 #include <net/checksum.h>
 #include <net/sock.h>
