--- ./vmblock-only/linux/inode.c	2017-07-06 11:47:09.496190133 -0600
+++ ./vmblock-only/linux/inode.c.new	2017-07-06 11:55:53.212410091 -0600
@@ -156,7 +156,7 @@
    inode->i_mode = S_IFLNK | S_IRWXUGO;
    inode->i_size = INODE_TO_IINFO(inode)->nameLen;
    inode->i_version = 1;
-   inode->i_atime = inode->i_mtime = inode->i_ctime = CURRENT_TIME;
+   inode->i_atime = inode->i_mtime = inode->i_ctime = current_time(inode);
 #if LINUX_VERSION_CODE < KERNEL_VERSION(3, 10, 0)
    inode->i_uid = inode->i_gid = 0;
 #else
