--- vmci-only/shared/vm_basic_defs.h	2017-06-19 20:29:40.000000000 -0600
+++ vmci-only/shared/vm_basic_defs.h.new	2017-07-06 12:11:37.146017203 -0600
@@ -720,6 +720,7 @@
                                                      lfMessageFont)
 
 /* This is not intended to be thread-safe. */
+#ifndef DO_ONCE
 #define DO_ONCE(code)                                                   \
    do {                                                                 \
       static Bool _doOnceDone = FALSE;                                  \
@@ -728,6 +729,7 @@
          code;                                                          \
       }                                                                 \
    } while (0)
+#endif
 
 /*
  * Bug 827422 and 838523.
