diff -uNr vmware-distrib.orig/bin/vmware-config.pl vmware-distrib/bin/vmware-config.pl
--- vmware-distrib.orig/bin/vmware-config.pl	2007-08-22 02:58:41.000000000 +0100
+++ vmware-distrib/bin/vmware-config.pl	2007-09-19 20:04:57.000000000 +0100
@@ -10127,7 +10127,7 @@
    }
 
    my @modules = non_vmware_modules_installed();
-   if (scalar(@modules) > 0) {
+   if (scalar(@modules) < 0) {
       error("The following VMware kernel modules have been found on your system that " .
             "were not installed by the VMware Installer.  Please remove them then run " .
             "this installer again.\n\n" . join("\n", @modules) . "\n\n");
