diff --git a/installer/services.sh b/installer/services.sh
index f883c2f..7256369 100755
--- a/installer/services.sh
+++ b/installer/services.sh
@@ -907,8 +907,13 @@ is_vmci_needed() {
 check_configured() {
    if [ -e "$vmware_etc_dir"/not_configured ]; then
       echo "`vmware_product_name`"' is installed, but it has not been (correctly) configured'
-      echo 'for the running kernel. To (re-)configure it, invoke the'
-      echo 'following command: '"$vmdb_answer_BINDIR"'/vmware-config.pl.'
+      echo 'for the running kernel.'
+      echo
+      echo 'Please ensure that the modules have been compiled for this kernel:'
+      echo '  emerge --oneshot vmware-modules'
+      echo
+      echo 'Also ensure '"`vmware_product_name`"' has been configured:'
+      echo '  '"$vmdb_answer_BINDIR"'/vmware-config.pl'
       echo
 
       exit 1
