diff --git a/bin/vmware-config.pl b/bin/vmware-config.pl
index bb9d04f..458a14b 100755
--- a/bin/vmware-config.pl
+++ b/bin/vmware-config.pl
@@ -10983,7 +10983,7 @@ sub main {
 
    if ($gOption{'tools-switch'} == 0) {
       my @modules = non_vmware_modules_installed();
-      if (scalar(@modules) > 0) {
+      if (scalar(@modules) < 0) {
          my $osVersion = direct_command(shell_string($gHelper{'uname'}) . ' -r');
          chomp($osVersion);
 	  error("The following VMware kernel modules have been found on your " .
