diff -ur vmware-distrib.orig/bin/vmware-config.pl vmware-distrib/bin/vmware-config.pl
--- vmware-distrib.orig/bin/vmware-config.pl	2006-03-24 23:13:05.000000000 +0100
+++ vmware-distrib/bin/vmware-config.pl	2006-03-24 23:26:40.000000000 +0100
@@ -1969,7 +1969,7 @@
     . ' -E - | ' . shell_string($gHelper{'grep'}) . ' '
     . shell_string($pattern));
   chomp($header_page_offset);
-  $header_page_offset =~ s/^$pattern \(?0x([0-9a-fA-F]{8,}).*$/$1/;
+  $header_page_offset =~ s/^$pattern \(?(\([^)]*\))?\)?0x([0-9a-fA-F]{8,}).*$/$2/;
   if ($header_page_offset =~ /[0-9a-fA-F]{8,}/) {
     # We found a valid page offset
     if (defined($gSystem{'page_offset'}) and
