--- vmware-distrib/bin/vmware-config.pl.ori	2005-11-09 01:45:08.000000000 +0000
+++ vmware-distrib/bin/vmware-config.pl	2005-11-09 01:48:21.000000000 +0000
@@ -8433,7 +8433,7 @@
          error('Please shut down any running VMs and run this script again.' .
                "\n\n");
       } else {
-         if (system(shell_string(db_get_answer('INITSCRIPTSDIR') . '/vmware')
+         if (system(shell_string('/etc/init.d' . '/vmware')
                     . ' stop')) {
             error('Unable to stop services for ' . vmware_product_name() .
                   "\n\n");
@@ -8463,8 +8463,8 @@
   }
   if (vmware_product() ne 'server') {
     configure_gtk2();
-    createMimePackageFile();
-    configureDesktopFiles();
+#   createMimePackageFile();
+#   configureDesktopFiles();
     configure_mon();
     configure_pp();
     configure_net();
@@ -8532,7 +8532,7 @@
   db_save();
   # Then start VMware's services
   if (!$gOption{'skip-stop-start'}) {
-    system(shell_string(db_get_answer('INITSCRIPTSDIR') . '/vmware') . ' start');
+    system(shell_string('/etc/init.d' . '/vmware') . ' start');
     print "\n";
   }
 
