diff --git a/modules/linux/vmhgfs/page.c b/modules/linux/vmhgfs/page.c
index 4fc9b53..d4a5d26 100644
--- a/modules/linux/vmhgfs/page.c
+++ b/modules/linux/vmhgfs/page.c
@@ -864,7 +864,15 @@ HgfsWriteBegin(struct file *file,             // IN: File to be written
    unsigned pageTo = pos + len;
    struct page *page;
 
+   /* 
+    * AOP_FLAG_NOFS was defined in the same changeset that 
+    * grab_cach_page_write_bring() was introduced.
+    */
+#ifdef AOP_FLAG_NOFS
+   page = grab_cache_page_write_begin(mapping, index, flags);
+#else
    page = __grab_cache_page(mapping, index);
+#endif
    if (page == NULL) {
       return -ENOMEM;
    }
