diff --git a/scripts/linux/resume-vm-default b/scripts/linux/resume-vm-default
index cebbba0..dbd38a6 100644
--- a/scripts/linux/resume-vm-default
+++ b/scripts/linux/resume-vm-default
@@ -25,41 +25,10 @@
 echo `date` ": Executing '$0'"
 echo
 
-find_networking_script() {
-    local script="error"
-    for dir in "/etc/init.d" "/sbin/init.d" "/etc" "/etc/rc.d" ; do
-        if [ -d "$dir/rc0.d" ] &&
-	    [ -d "$dir/rc1.d" ] &&
-	    [ -d "$dir/rc2.d" ] &&
-	    [ -d "$dir/rc3.d" ] &&
-	    [ -d "$dir/rc4.d" ] &&
-	    [ -d "$dir/rc5.d" ] &&
-	    [ -d "$dir/rc6.d" ]; then
-
-	    # Now find the appropriate networking script.
-	    if [ -d "$dir/init.d" ]; then
-		if [ -x "$dir/init.d/network" ]; then
-		    script="$dir/init.d/network"
-		elif [ -x "$dir/init.d/networking" ]; then
-		    script="$dir/init.d/networking"
-		fi
-	    else
-		if [ -x "$dir/network" ]; then
-		    script="$dir/network"
-		elif [ -x "$dir/networking" ]; then
-		    script="$dir/networking"
-		fi
-	    fi
-        fi
-    done
-
-    echo "$script"
-}
-
 rescue_NIC() {
    niclist="/var/run/vmware-active-nics"
 
-   ifup_path=`which ifup 2>/dev/null`;
+   ifup_path="/etc/init.d/net.";
    if [ $? -ne 0 ]; then
       return 1;
    fi
@@ -76,7 +45,7 @@ rescue_NIC() {
          else
             echo `date` "[rescue_nic] activating $nic ..."
 
-            $ifup_path $nic
+            $ifup_path$nic restart
          fi
       done < $niclist
 
@@ -95,15 +64,8 @@ if [ -d "$scriptsdir" ]; then
     done
 fi
 
-network=`find_networking_script`
-if [ "$network" != "error" ]; then
-   "$network" restart
-   # Continue even if the networking init script wasn't successful.
-   status=0
-else
-   echo "networking script not found"
-   status=1
-fi
+# Continue even if the networking init script wasn't successful.
+status=0
 
 if [ $status -eq 0 ]; then
     rescue_NIC
diff --git a/scripts/linux/suspend-vm-default b/scripts/linux/suspend-vm-default
index 80bf5b9..f590194 100644
--- a/scripts/linux/suspend-vm-default
+++ b/scripts/linux/suspend-vm-default
@@ -25,37 +25,6 @@
 echo `date` ": Executing '$0'"
 echo
 
-find_networking_script() {
-    local script="error"
-    for dir in "/etc/init.d" "/sbin/init.d" "/etc" "/etc/rc.d" ; do
-        if [ -d "$dir/rc0.d" ] &&
-	    [ -d "$dir/rc1.d" ] &&
-	    [ -d "$dir/rc2.d" ] &&
-	    [ -d "$dir/rc3.d" ] &&
-	    [ -d "$dir/rc4.d" ] &&
-	    [ -d "$dir/rc5.d" ] &&
-	    [ -d "$dir/rc6.d" ]; then
-
-	    # Now find the appropriate networking script.
-	    if [ -d "$dir/init.d" ]; then
-		if [ -x "$dir/init.d/network" ]; then
-		    script="$dir/init.d/network"
-		elif [ -x "$dir/init.d/networking" ]; then
-		    script="$dir/init.d/networking"
-		fi
-	    else
-		if [ -x "$dir/network" ]; then
-		    script="$dir/network"
-		elif [ -x "$dir/networking" ]; then
-		    script="$dir/networking"
-		fi
-	    fi
-        fi
-    done
-
-    echo "$script"
-}
-
 save_active_NIC_list() {
     ifconfig_path=`which ifconfig 2>/dev/null`
     if [ $? ]; then
@@ -76,15 +45,12 @@ fi
 
 save_active_NIC_list
 
-network=`find_networking_script`
-if [ "$network" != "error" ]; then
-   "$network" stop
-   # If the network is down, this may fail but that's not a good reason
-   # to prevent the suspend.
-   status=0
-else
-   echo "networking script not found"
-   status=1
-fi
+for network in `ls /etc/init.d/net.*`; 
+do
+  "$network" stop
+done
+# If the network is down, this may fail but that's not a good reason
+# to prevent the suspend.
+status=0
 
 exit "$status"
