diff -uNr open-vm-tools-2008.01.23-74039.old/configure.ac open-vm-tools-2008.01.23-74039/configure.ac
--- open-vm-tools-2008.01.23-74039.old/configure.ac	2008-02-05 12:41:50.000000000 +0000
+++ open-vm-tools-2008.01.23-74039/configure.ac	2008-02-05 12:46:42.000000000 +0000
@@ -256,7 +256,7 @@
    [crypt],
    [crypt],
    [HAVE_CRYPT="yes"
-    GUESTD_LDFLAGS="$GUESTD_LDFLAGS -lcrypt"],
+    GUESTD_LDADD="$GUESTD_LDADD -lcrypt"],
    [AC_MSG_ERROR(
       [libcrypt not found. Please install the libc/libcrypt devel package(s).])])
 
@@ -266,7 +266,7 @@
    [AC_CHECK_LIB(
       dl,
       dlopen,
-      GUESTD_LDFLAGS="$GUESTD_LDFLAGS -ldl",
+      GUESTD_LDADD="$GUESTD_LDADD -ldl",
       [AC_MSG_ERROR(
          [dlopen was not found, but is required for open-vm-tools to function properly. Please contact your OS vendor.])])])
 
@@ -276,8 +276,8 @@
 AC_CHECK_LIB(
    [pthread],
    [pthread_mutex_init],
-   [GUESTD_LDFLAGS="$GUESTD_LDFLAGS -lpthread"
-    VMWARE_USER_LDFLAGS="$VMWARE_USER_LDFLAGS -lpthread"],
+   [GUESTD_LDADD="$GUESTD_LDADD -lpthread"
+    VMWARE_USER_LDADD="$VMWARE_USER_LDADD -lpthread"],
    [AC_MSG_ERROR(
       [libpthread not found. Please install the libc/libpthread devel package(s).])])
 
@@ -297,8 +297,8 @@
    DNET_CFLAGS="`$DNET_CONFIG --cflags`"
    DNET_LIBS="`$DNET_CONFIG --libs`"
 
-   GUESTD_LDFLAGS="$GUESTD_LDFLAGS $DNET_LIBS"
-   VMWARE_USER_LDFLAGS="$VMWARE_USER_LDFLAGS $DNET_LIBS"
+   GUESTD_LDADD="$GUESTD_LDADD $DNET_LIBS"
+   VMWARE_USER_LDADD="$VMWARE_USER_LDADD $DNET_LIBS"
 fi
 
 ###
@@ -490,14 +490,14 @@
    LIB_IMPERSONATE_CFLAGS="$LIB_IMPERSONATE_CFLAGS -D_POSIX_PTHREAD_SEMANTICS"
    LIB_USER_CFLAGS="$LIB_USER_CFLAGS -D_POSIX_PTHREAD_SEMANTICS"
 
-   GUESTD_LDFLAGS="$GUESTD_LDFLAGS -lsocket"
-   GUESTD_LDFLAGS="$GUESTD_LDFLAGS -lnsl"
+   GUESTD_LDADD="$GUESTD_LDADD -lsocket"
+   GUESTD_LDADD="$GUESTD_LDADD -lnsl"
 
    VMWARE_USER_CFLAGS="$VMWARE_USER_CFLAGS -I/usr/X11/include/"
-   VMWARE_USER_LDFLAGS="$VMWARE_USER_LDFLAGS -lXext"
-   VMWARE_USER_LDFLAGS="$VMWARE_USER_LDFLAGS -lX11"
-   VMWARE_USER_LDFLAGS="$VMWARE_USER_LDFLAGS -lsocket"
-   VMWARE_USER_LDFLAGS="$VMWARE_USER_LDFLAGS -lnsl"
+   VMWARE_USER_LDADD="$VMWARE_USER_LDADD -lXext"
+   VMWARE_USER_LDADD="$VMWARE_USER_LDADD -lX11"
+   VMWARE_USER_LDADD="$VMWARE_USER_LDADD -lsocket"
+   VMWARE_USER_LDADD="$VMWARE_USER_LDADD -lnsl"
 
    # uname -r on Solaris 10 reports "5.10"
    if test "$osVersion" -ge 510000; then
@@ -571,10 +571,10 @@
 
 ### Program substs
 
-AC_SUBST([GUESTD_LDFLAGS])
+AC_SUBST([GUESTD_LDADD])
 
 AC_SUBST([VMWARE_USER_CFLAGS])
-AC_SUBST([VMWARE_USER_LDFLAGS])
+AC_SUBST([VMWARE_USER_LDADD])
 
 ###
 ### Create the Makefiles
diff -uNr open-vm-tools-2008.01.23-74039.old/guestd/Makefile.am open-vm-tools-2008.01.23-74039/guestd/Makefile.am
--- open-vm-tools-2008.01.23-74039.old/guestd/Makefile.am	2008-02-05 12:41:50.000000000 +0000
+++ open-vm-tools-2008.01.23-74039/guestd/Makefile.am	2008-02-05 12:51:08.000000000 +0000
@@ -77,9 +77,7 @@
    HGFS_BD =
 endif
 guestd_LDADD += $(HGFS_BD)
-
-guestd_LDFLAGS =
-guestd_LDFLAGS += @GUESTD_LDFLAGS@
+guestd_LDADD += @GUESTD_LDADD@
 
 guestd_SOURCES =
 guestd_SOURCES += debugStdio.c
diff -uNr open-vm-tools-2008.01.23-74039.old/toolbox/Makefile.am open-vm-tools-2008.01.23-74039/toolbox/Makefile.am
--- open-vm-tools-2008.01.23-74039.old/toolbox/Makefile.am	2008-02-05 12:41:50.000000000 +0000
+++ open-vm-tools-2008.01.23-74039/toolbox/Makefile.am	2008-02-05 12:53:55.000000000 +0000
@@ -50,10 +50,8 @@
 toolbox_LDADD += ../lib/misc/libMisc.a
 toolbox_LDADD += ../lib/panicDefault/libPanicDefault.a
 toolbox_LDADD += ../lib/panic/libPanic.a
-
-toolbox_LDFLAGS =
-toolbox_LDFLAGS += -lX11
-toolbox_LDFLAGS += @GTK_LIBS@
+toolbox_LDADD += -lX11
+toolbox_LDADD += @GTK_LIBS@
 
 toolbox_SOURCES =
 toolbox_SOURCES += debugStdio.c
diff -uNr open-vm-tools-2008.01.23-74039.old/vmware-user/Makefile.am open-vm-tools-2008.01.23-74039/vmware-user/Makefile.am
--- open-vm-tools-2008.01.23-74039.old/vmware-user/Makefile.am	2008-02-05 12:41:50.000000000 +0000
+++ open-vm-tools-2008.01.23-74039/vmware-user/Makefile.am	2008-02-05 12:50:33.000000000 +0000
@@ -67,12 +67,10 @@
 vmware_user_LDADD += ../lib/auth/libAuth.a
 vmware_user_LDADD += ../lib/panicDefault/libPanicDefault.a
 vmware_user_LDADD += ../lib/panic/libPanic.a
-
-vmware_user_LDFLAGS =
-vmware_user_LDFLAGS += -lcrypt
-vmware_user_LDFLAGS += @GTK_LIBS@
-vmware_user_LDFLAGS += @VMWARE_USER_LDFLAGS@
-vmware_user_LDFLAGS += @COMMON_XLIBS@
+vmware_user_LDADD += -lcrypt
+vmware_user_LDADD += @GTK_LIBS@
+vmware_user_LDADD += @VMWARE_USER_LDADD@
+vmware_user_LDADD += @COMMON_XLIBS@
 
 vmware_user_SOURCES =
 vmware_user_SOURCES += copyPaste.c
