diff --git a/payload/install/vmware-installer/vmis/__init__.py b/payload/install/vmware-installer/vmis/__init__.py
index 7a46312..e1dbff1 100644
--- a/payload/install/vmware-installer/vmis/__init__.py
+++ b/payload/install/vmware-installer/vmis/__init__.py
@@ -12,5 +12,5 @@ VERSION_INFO = (1,0)
 VERSION = '.'.join([str(x) for x in VERSION_INFO])
 MAJOR_VERSION = VERSION_INFO[0]
 
-CONFDIR = path(u'/etc/vmware')
+CONFDIR = path(u'./vmware-confdir')
 DATABASE_PATH = CONFDIR/'database'
diff --git a/payload/install/vmware-installer/vmis/core/questions.py b/payload/install/vmware-installer/vmis/core/questions.py
index 2be10de..f92ecff 100644
--- a/payload/install/vmware-installer/vmis/core/questions.py
+++ b/payload/install/vmware-installer/vmis/core/questions.py
@@ -139,6 +139,9 @@ class InitDir(Directory):
       """
       super(InitDir, self).Validate(answer)
 
+      ### GENTOO PATCH ###  Finish early
+      return True
+
       rcdirs = ('rc0.d', 'rc1.d', 'rc2.d', 'rc3.d', 'rc4.d', 'rc5.d', 'rc6.d')
       answer = path(answer)
 
diff --git a/payload/install/vmware-installer/vmis/ui/console.py b/payload/install/vmware-installer/vmis/ui/console.py
index 525df75..abb81b3 100644
--- a/payload/install/vmware-installer/vmis/ui/console.py
+++ b/payload/install/vmware-installer/vmis/ui/console.py
@@ -148,7 +148,7 @@ class Wizard(object):
          wrapper = TextWrapper()
          wrapper.width = 79
          wrapper.replace_whitespace = False # Needed to preserve paragraph spacing.
-         Popen('more', stdin=PIPE).communicate(input=wrapper.fill(text))
+         Popen('less', stdin=PIPE).communicate(input=wrapper.fill(text))
       except IOError: # RHEL4 appears to close stdin while we still expect it to be open
          pass
 
diff --git a/payload/install/vmware-installer/vmis/util/log.py b/payload/install/vmware-installer/vmis/util/log.py
index 4f609c4..b07b89d 100644
--- a/payload/install/vmware-installer/vmis/util/log.py
+++ b/payload/install/vmware-installer/vmis/util/log.py
@@ -22,6 +22,8 @@ else:
    # running a build)
    LOG_FILE = '/tmp/vmware-installer.log'
 
+LOG_FILE = './vmware-installer.log'
+
 if level <= logging.DEBUG:
    MAX_BYTES = 0
    BACKUP_COUNT = 0
