--- vmware/bin/vmware	2004-12-02 07:12:26.268618096 +0900
+++ vmware/bin/vmware	2004-12-02 07:34:31.684433747 +0900
@@ -169,15 +169,23 @@
    local lib
    local dummy
    local status
+   local newpath
 
    path=''
    while read -r lib dummy status; do
       if [ "$status" = 'not found' ]; then
+         newpath="$vm_db_answer_LIBDIR"'/lib/'"$lib"
          if [ "$path" = '' ]; then
-            path="$vm_db_answer_LIBDIR"'/lib/'"$lib"
+	    path="$newpath"
          else
-            path="$path"':'"$vm_db_answer_LIBDIR"'/lib/'"$lib"
+            path="$path"':'"$newpath"
          fi
+	 for x in "$newpath/"*.so; do
+	    if [ -f "$x" ]
+	    then
+	       path="$path"':'`LD_LIBRARY_PATH="$path" LANGUAGE=C LANG=C ldd "$x" | vm_append_libs`
+            fi
+         done
       fi
    done
 
