diff -uNr vmware-server-distrib.orig/bin/vmware-config.pl vmware-server-distrib/bin/vmware-config.pl
--- vmware-server-distrib.orig/bin/vmware-config.pl	2007-12-02 11:40:41.000000000 +0000
+++ vmware-server-distrib/bin/vmware-config.pl	2007-12-02 11:43:29.000000000 +0000
@@ -10162,7 +10162,7 @@
          error('Please shut down any running VMs and run this script again.' .
                "\n\n");
       } else {
-         if (system(shell_string(db_get_answer('INITSCRIPTSDIR') . '/vmware')
+         if (system(shell_string('/etc/init.d' . '/vmware')
                     . ' stop')) {
             error('Unable to stop services for ' . vmware_product_name() .  "\n\n");
          }
@@ -10212,8 +10212,8 @@
  
   if (isDesktopProduct()) {
     build_vmblock();
-    createMimePackageFile();
-    configureDesktopFiles();
+    # createMimePackageFile();
+    # configureDesktopFiles();
     configure_eclipse_plugin();
   }
 
@@ -10283,7 +10283,7 @@
   db_save();
   # Then start VMware's services
   if (!$gOption{'skip-stop-start'}) {
-    system(shell_string(db_get_answer('INITSCRIPTSDIR') . '/vmware') . ' start');
+    system(shell_string('/etc/init.d' . '/vmware') . ' start');
     print "\n";
   }
 
