diff -uNr vmware-server-distrib.orig/bin/vmware-config.pl vmware-server-distrib/bin/vmware-config.pl
--- vmware-server-distrib.orig/bin/vmware-config.pl	2007-12-02 11:16:42.000000000 +0000
+++ vmware-server-distrib/bin/vmware-config.pl	2007-12-02 11:25:36.000000000 +0000
@@ -9097,7 +9097,10 @@
     error('Unable to write configuration file "' . $name . '".' . "\n\n");
   }
   db_add_file($name, $cFlagTimestamp | $cFlagConfig);
-  safe_chmod(0644, $name);
+  safe_chmod(0664, $name);
+  my $gid = (getgrnam('vmware'))[2];
+  my $uid = (stat($name))[4];
+  safe_chown($uid,$gid,$name);
 
   # Append the promotional configuration if it exists
   $promoconfig = $libdir . '/configurator/PROMOCONFIG';
