diff -uNr vmware-tools-distrib.orig/bin/vmware-config-tools.pl vmware-tools-distrib/bin/vmware-config-tools.pl
--- vmware-tools-distrib.orig/bin/vmware-config-tools.pl	2007-05-02 04:36:55.000000000 +0100
+++ vmware-tools-distrib/bin/vmware-config-tools.pl	2007-07-11 23:24:16.000000000 +0100
@@ -3551,11 +3551,15 @@
     $mod_dir = db_get_answer('LIBDIR') . '/modules';
   }
 
+  $gOption{'compile'} = 1;
+  return 'yes';
+
   if (($name eq 'vmxnet') and (not is64BitKernel())) {
     # Figure out the correct network script.
     my $init_dir = db_get_answer('INITSCRIPTSDIR');
     my $network_path = find_first_exist("$init_dir/network",
-                                        "$init_dir/networking");
+                                        "$init_dir/networking",
+                                        "/bin/false");
     if (!defined($network_path)) {
       print wrap("Can not find $init_dir/network and $init_dir/networking.\n\n", 0);
       return 'no';
@@ -9238,7 +9242,7 @@
   #
   print "\n";
   if (!$gOption{'skip-stop-start'} &&
-      system(shell_string(db_get_answer('INITSCRIPTSDIR') . $vmwareToolsScript)
+      system(shell_string('/etc/init.d/' . $vmwareToolsScript)
              . ' stop')) {
     print wrap('Making sure services for ' . vmware_product_name()
                 . ' are stopped.' . "\n\n", 0);
@@ -9279,7 +9283,7 @@
  # configure the experimental drivers
  if ( vmware_product() eq 'tools-for-linux') {
     configure_vmdesched();
-    configure_vmci();
+ #   configure_vmci();
  }
 
   write_tools_config();
@@ -9287,9 +9291,9 @@
   if (vmware_product() eq 'tools-for-linux') {
     configure_thinprint();
   }
-  $result = configure_X();
-  $gFeatureInstallResults[$cFeatureInstallIDs{'xserver'}] =
-    $result eq 'no' ?  $cFeatureInstallFailed : $cFeatureInstalled;
+  #$result = configure_X();
+  $gFeatureInstallResults[$cFeatureInstallIDs{'xserver'}] = $cFeatureInstalled;
+  #  $result eq 'no' ?  $cFeatureInstallFailed : $cFeatureInstalled;
 
   if (vmware_product() eq 'tools-for-linux' ||
       vmware_product() eq 'tools-for-solaris') {
@@ -9303,7 +9307,7 @@
   # Then start VMware's services.
   #
   if (!$gOption{'skip-stop-start'} &&
-      system(shell_string(db_get_answer('INITSCRIPTSDIR') . $vmwareToolsScript)
+      system(shell_string('/etc/init.d/' . $vmwareToolsScript)
              . ' start')) {
     error('Unable to start services for ' . vmware_product_name() . "\n\n");
   } else {
@@ -9343,7 +9347,7 @@
 
   my $network_path = find_first_exist("/etc/init.d/network",
                                       "/etc/init.d/networking");
-  if (vmware_product() eq 'tools-for-linux') {
+  if (vmware_product() eq 'tools-for-linux-off') {
     # Build dependency data for the new modules so the
     # system knows they're there and can find them.
     system(shell_string($gHelper{'depmod'}) . ' -a');
diff -uNr vmware-tools-distrib.orig/installer/services.sh vmware-tools-distrib/installer/services.sh
--- vmware-tools-distrib.orig/installer/services.sh	2007-05-02 04:36:57.000000000 +0100
+++ vmware-tools-distrib/installer/services.sh	2007-07-11 22:33:23.000000000 +0100
@@ -663,7 +663,7 @@
    if [ "`isLoaded "$1"`" = 'yes' ]; then
       /sbin/rmmod "$1"
    fi
-   /sbin/insmod -s -f "/lib/modules/`uname -r`/misc/$1.o" >/dev/null 2>&1 || /sbin/insmod -s -f "$1" >/dev/null 2>&1 || exit 1
+   /sbin/modprobe $1 || /sbin/insmod -s -f "/lib/modules/`uname -r`/misc/$1.o" >/dev/null 2>&1 || /sbin/insmod -s -f "$1" >/dev/null 2>&1 || exit 1
    return 0
 }
 
