diff -urN vmware-tools-distrib.orig/bin/vmware-config-tools.pl vmware-tools-distrib/bin/vmware-config-tools.pl
--- vmware-tools-distrib.orig/bin/vmware-config-tools.pl	2006-06-06 07:10:53.000000000 -0400
+++ vmware-tools-distrib/bin/vmware-config-tools.pl	2006-06-07 11:49:07.000000000 -0400
@@ -3164,7 +3164,8 @@
     # Figure out the correct network script.
     my $init_dir = db_get_answer('INITSCRIPTSDIR');
     my $network_path = find_first_exist("$init_dir/network",
-                                        "$init_dir/networking");
+                                        "$init_dir/networking",
+					"/etc/init.d/net.eth0");
     if (!defined($network_path)) {
       print wrap("Can not find $init_dir/network and $init_dir/networking.\n\n", 0);
       return 'no';
@@ -7925,6 +7926,7 @@
   uninstall_file($gConfFlag);
 
   db_save();
+
   #
   # Then start VMware's services.
   #
@@ -7946,7 +7948,8 @@
              0);
 
   my $network_path = find_first_exist("/etc/init.d/network",
-                                      "/etc/init.d/networking");
+                                      "/etc/init.d/networking",
+				      "/etc/init.d/net.eth0");
   if (vmware_product() eq 'tools-for-linux' and
       db_get_answer('VMXNET_CONFED') eq 'yes') {
     print wrap('To use the vmxnet driver, restart networking using the '
