diff -uNr vmhgfs-only.orig/driver.c vmhgfs-only/driver.c
--- vmhgfs-only.orig/driver.c	2007-03-05 01:46:22.000000000 +0000
+++ vmhgfs-only/driver.c	2007-03-05 01:47:10.000000000 +0000
@@ -760,7 +760,9 @@
    inode->i_uid = 0;   /* This is bogus, should be the mount owner. */
    inode->i_gid = 0;   /* This is bogus, should be the mount owner. */
    inode->i_rdev = 0;  /* Device nodes are not supported */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 18)
    inode->i_blksize = HGFS_BLOCKSIZE;
+#endif
    inode->i_blocks = (attr->size + HGFS_BLOCKSIZE - 1) / HGFS_BLOCKSIZE;
    inode->i_size = attr->size;
    HGFS_SET_TIME(inode->i_atime, attr->accessTime);
