diff -Naurd original/vmhgfs-only/driver.c modified/vmhgfs-only/driver.c
--- original/vmhgfs-only/driver.c	2007-02-27 03:46:41.000000000 -0600
+++ modified/vmhgfs-only/driver.c	2007-02-27 03:30:06.000000000 -0600
@@ -20,6 +20,10 @@
 #include "hgfsDevLinux.h"  // for HgfsMountInfo definition
 #include "hgfsProto.h"     // for hgfs protocol
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 18)
+#include <linux/mount.h>
+#endif
+
 #include <linux/vfs.h>
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2, 5, 25)
@@ -41,8 +45,13 @@
 #define HGFS_SB_TO_COMMON(sb)             ((HgfsSuperInfo *)(sb)->s_fs_info)
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 19)
+#define INODE_SET_II_P(inode, info) do { (inode)->i_private = (info); } while (0)
+#define INODE_GET_II_P(inode) ((HgfsInodeInfo *)(inode)->i_private)
+#else
 #define INODE_SET_II_P(inode, info) do { (inode)->u.generic_ip = (info); } while (0)
 #define INODE_GET_II_P(inode) ((HgfsInodeInfo *)(inode)->u.generic_ip)
+#endif
 
 /*
  * 2.5.x kernels support nanoseconds timestamps.
