diff -uNr vmware-server-distrib.orig/bin/vmware-config.pl vmware-server-distrib/bin/vmware-config.pl
--- vmware-server-distrib.orig/bin/vmware-config.pl	2006-02-18 18:12:26.000000000 +0000
+++ vmware-server-distrib/bin/vmware-config.pl	2006-02-18 18:14:17.000000000 +0000
@@ -8571,10 +8571,10 @@
     configure_gtk2();
 #   createMimePackageFile();
 #   configureDesktopFiles();
-    configure_mon();
-    configure_pp();
+#   configure_mon();
+#   configure_pp();
     configure_net();
-    build_vmnet();
+#    build_vmnet();
   }
 
   # Create the directory for the UNIX domain sockets
