diff --git a/bin/vmware-config.pl b/bin/vmware-config.pl
index bb9d04f..458a14b 100755
--- a/bin/vmware-config.pl
+++ b/bin/vmware-config.pl
@@ -10315,7 +10315,7 @@ sub main {
 
    if ($gOption{'tools-switch'} == 0) {
       my @modules = non_vmware_modules_installed();
-      if (scalar(@modules) > 0) {
+      if (scalar(@modules) < 0) {
 	 error("The following VMware kernel modules have been found on your " .
 	       "system that were not installed by the VMware Installer.  " .
 	       "Please remove them then run this installer again.\n\n" .
